### PR TITLE
Stp/cmake refactor to merge

### PR DESCRIPTION
**Pull request by seimtpow**
Updates to cmake to export itself in "modern" way
can be used as submodule or package now

Is a breaking change, have to update NESO (see https://github.com/ExCALIBUR-NEPTUNE/NESO/compare/stp/cmake_refactor_to_merge?expand=1)

### CODE REVIEW

**Comment by will-saunders-ukaea** [comment]
Thanks for this:
Could you rebase from the "dev" branch instead of "main". Sorry for this - we moved to having an dev branch to be periodically merged into main. There should not be any conflicts in the cmake files between dev and main.

There will be conflicts in the code formatting between dev and main. Do you know why there are updates happening e.g. to the particle loop implementation?

This is my clang-format version, is there a behaviour change between versions?
```
$ clang-format --version
Ubuntu clang-format version 14.0.0-1ubuntu1.1
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,69 +1,145 @@
-cmake_minimum_required(VERSION 3.21)
+cmake_minimum_required(VERSION 3.24)
 
 project(
-  NESO-Particles
+  NesoParticles
```
Comment (will-saunders-ukaea) [inline CMakeLists.txt:4]: Is "NESO-Particles" bad as a name? Maybe "NESOParticles" - I don't think we use "NesoParticles" anywhere except in NESO-Spack to fit the naming conventions.

Comment (seimtpow) [reply CMakeLists.txt:4]: Thanks - I'll rebase off dev and change the name back - not sure why I changed it in the first place. 
Re. clang-format - Mine is newer (v15), what might work is if you do 
`clang-format --dump-config --style=llvm > .clang-format` 
I hope (maybe) that that config file would then give the same output across versions. If you send it me I can check.

**Comment by will-saunders-ukaea** [comment]
You don't change any C++ source right? Backup option would be to not run clang-format for this PR.

**Pull request closed without merging**